### PR TITLE
Bugfix/pelagos 5476 prevent udi re use

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Util/Udi.php [hunk 1]
```diff
@@ -85,8 +86,18 @@ public function mintUdi(Dataset $dataset)
...
-        // Append the sequence to the UDI.
-        $udi .= sprintf('%04d', $sequence);
+        // Test for the edge case where UDI is in the Deleted UDI list.
+        // This can only happen if the dataset with the highest sequence gets
+        // deleted.
+        $testUdi = $udi . sprintf('%04d', $sequence);
+        $deletedUdi = $this->entityManager->getRepository(DeletedUdi::class)->findOneBy(['udi' => $testUdi]);
+        if ($deletedUdi) {
+            // Incremenent the sequence again to make it unique. (logically there is no need to re-check)
+            $udi .= sprintf('%04d', $sequence + 1);
```
[the remark this answers is not in the document]
Comment (fightingtexasaggie) [reply src/Util/Udi.php:96]: Indeed this could happen if more than one are deleted off the top end of the sequence. I think I like your idea of instead recording every UDI issued (and obviously backporting the deleted UDIs). The minter would then do this check recursively until it doesn't find an udi already issued.
